cli/lsp/resolver.rs

```diff
@@ -50,34 +52,39 @@ use std::path::Path;
 use std::rc::Rc;
 use std::sync::Arc;
 
+use super::documents::Document;
+
 #[derive(Debug, Clone)]
 pub struct LspResolver {
+  cache: Arc<dyn HttpCache>,
   graph_resolver: Arc<CliGraphResolver>,
-  jsr_resolver: Option<Arc<JsrCacheResolver>>,
+  jsr_resolver: Arc<JsrCacheResolver>,
   npm_resolver: Option<Arc<dyn CliNpmResolver>>,
   node_resolver: Option<Arc<CliNodeResolver>>,
   npm_config_hash: LspNpmConfigHash,
   redirect_resolver: Option<Arc<RedirectResolver>>,
   graph_imports: Arc<IndexMap<ModuleSpecifier, GraphImport>>,
   config: Arc<Config>,
+  unstable_sloppy_imports: bool,
 }
 
-impl Default for LspResolver {
-  fn default() -> Self {
+impl LspResolver {
+  pub fn new(cache: Arc<dyn HttpCache>) -> Self {
```

Comment:
I've been trying to avoid passing cache here, since the first value we pass in during `Inner::new()` isn't correct (doesn't respect workspace settings) and shouldn't be used. Been meaning to do the same for `Documents` as well. Though it would internally default to the same thing via `LspCache::new(None)`. Just a heads up, I can deal with it later.

Reply:
I was looking at the code more and I think passing `cache` isn't necessary. I'll remove it.

Reply:
Reverted in https://github.com/denoland/deno/pull/23751/commits/1d8e394c6c39f4e62a3aeed54556b31428d32434